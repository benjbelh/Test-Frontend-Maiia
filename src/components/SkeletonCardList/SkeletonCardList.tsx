import React, { FC } from 'react';
import SkeletonCard from 'components/SkeletonCard/SkeletonCard';

type SkeletonCardListProps = {
  cardNumber: number;
};

export const SkeletonCardList: FC<SkeletonCardListProps> = ({ cardNumber = 4 }) => (
  <>
    {[...Array(cardNumber)].map((item, index) => (
      <SkeletonCard key={index} />
    ))}
  </>
);

export default SkeletonCardList;
