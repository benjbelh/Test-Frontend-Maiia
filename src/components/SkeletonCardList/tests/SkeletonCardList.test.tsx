import React from 'react';
import { render } from '@testing-library/react';
import SkeletonCardList from '../SkeletonCardList';

describe('SkeletonCardList', () => {
  it('should renders correctly', () => {
    const { asFragment } = render(<SkeletonCardList cardNumber={6} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
