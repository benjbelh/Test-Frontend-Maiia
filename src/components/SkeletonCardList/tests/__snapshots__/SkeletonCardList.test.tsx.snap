// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonCardList should renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style="height: 250px;"
      />
      <div
        class="MuiCardContent-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
          style="width: 60%;"
        />
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style="width: 30%; height: 30px;"
      />
    </div>
  </div>
  <div
    class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style="height: 250px;"
      />
      <div
        class="MuiCardContent-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
          style="width: 60%;"
        />
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style="width: 30%; height: 30px;"
      />
    </div>
  </div>
  <div
    class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style="height: 250px;"
      />
      <div
        class="MuiCardContent-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
          style="width: 60%;"
        />
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style="width: 30%; height: 30px;"
      />
    </div>
  </div>
  <div
    class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style="height: 250px;"
      />
      <div
        class="MuiCardContent-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
          style="width: 60%;"
        />
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style="width: 30%; height: 30px;"
      />
    </div>
  </div>
  <div
    class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style="height: 250px;"
      />
      <div
        class="MuiCardContent-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
          style="width: 60%;"
        />
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style="width: 30%; height: 30px;"
      />
    </div>
  </div>
  <div
    class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
        style="height: 250px;"
      />
      <div
        class="MuiCardContent-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
          style="width: 60%;"
        />
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style="width: 30%; height: 30px;"
      />
    </div>
  </div>
</DocumentFragment>
`;
